### PR TITLE
Support Defining Ordering Equivalence at the Source

### LINKED ISSUE #6468: Defining Ordering Equivalence at the Source
### Is your feature request related to a problem or challenge?

Currently, we have the support of a table given an ordering such as `WITH ORDER (c ASC)` with `CREATE EXTERNAL TABLE` statement. However, if we need to give 2 or more orderings for a table, we cannot define it with multiple WITH ORDER's.

### Describe the solution you'd like

Output orderings are in the type of `Option<Vec<PhysicalSortExpr>>`. After we properly parse multiple with order's, we can refactor it as `Vec<Vec<PhysicalSortExpr>>`.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by berkaysynnada**
# Which issue does this PR close?

Closes #6468.

# Rationale for this change

If we try to assign 2 or more orderings to a table with `CREATE EXTERNAL TABLE`, we cannot express it with multiple WITH ORDER's. With this PR, we can create such tables like this:
```
CREATE EXTERNAL TABLE multiple_ordered_table (
...
 )
 STORED AS CSV
 WITH HEADER ROW
 WITH ORDER (a ASC, b ASC)
 WITH ORDER (c ASC)
 LOCATION '...';
```

# What changes are included in this PR?

Parser can handle multiple WITH ORDER options. In the codebase, output orderings are in the type of `Option<Vec<PhysicalSortExpr>>`. It is refactored as `Vec<Vec<PhysicalSortExpr>>`.

# Are these changes tested?

Yes, related slt tests are added showing the usage of feature on plans.

# Are there any user-facing changes?

The ListingOptions API is changed from 
`pub fn with_file_sort_order(mut self, file_sort_order: Option<Vec<Expr>>) -> Self ` to 
`pub fn with_file_sort_order(mut self, file_sort_order: Vec<Vec<Expr>>) -> Self`. This change was necessary to give all sort orderings at once.

### CODE REVIEW

**Review by mustafasrepo — APPROVED** [review]
This PR is passed from our internal review process. It is LTGM!.

**Review by alamb — COMMENTED** [review]
Looks like a nice change to me -- thank you @berkaysynnada  and @mustafasrepo

**Review by alamb — APPROVED** [review]

File: datafusion/core/src/datasource/listing/table.rs [hunk 1]
```diff
@@ -418,7 +421,7 @@ impl ListingOptions {
     ///
     /// assert_eq!(listing_options.file_sort_order, file_sort_order);
     /// ```
-    pub fn with_file_sort_order(mut self, file_sort_order: Option<Vec<Expr>>) -> Self {
+    pub fn with_file_sort_order(mut self, file_sort_order: Vec<Vec<Expr>>) -> Self {
```
[the remark this answers is not in the document]
Comment (berkaysynnada) [reply datafusion/core/src/datasource/listing/table.rs:424]: We have also considered such a design, and you can check our observations [here](https://github.com/synnada-ai/arrow-datafusion/pull/107#pullrequestreview-1443036721). In summary, that API cannot provide a reset for sort orders, and the user needs to call it multiple times for multiple orderings (that is not the case for other methods).

We have replaced `Vec<Vec<OrderByExpr>>` and `Vec<Vec<PhysicalSortExpr>>` with  `Vec<LexOrdering>` where the inner vector is used for a lexicographical sorting and the outer is used for another lex sorting.

**Comment by alamb** [comment]
I think everything looks good to me here -- thanks again @berkaysynnada and @mustafasrepo

File: datafusion/sql/src/parser.rs [hunk 2]
```diff
@@ -585,7 +588,7 @@ impl<'a> DFParser<'a> {
             delimiter: Option<char>,
             file_compression_type: Option<CompressionTypeVariant>,
             table_partition_cols: Option<Vec<String>>,
-            order_exprs: Option<Vec<OrderByExpr>>,
+            order_exprs: Vec<Vec<OrderByExpr>>,
```
[the remark this answers is not in the document]
Comment (berkaysynnada) [reply datafusion/sql/src/parser.rs]: I have replaced all `Vec<Vec<OrderByExpr>> `'s with `Vec<LexOrdering>`'s. Thanks for the review

**Pull request merged**